graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/DgsSchemaProviderTest.kt

```diff
@@ -157,6 +158,21 @@ internal class DgsSchemaProviderTest {
         assertEquals("location2-schema2.graphqls", schemaFiles[3].filename)
     }
 
+    @Test
+    fun `Should specify sourceName on SourceLocation`() {
+        val schemaProvider = schemaProvider(
+            schemaLocations = listOf("classpath*:schema/**/*.graphql*")
+        )
+
+        val schema = schemaProvider.schema()
+
+        for (type in schema.allTypesAsList) {
+            if (type.definition?.sourceLocation != null) {
```

Comment:
Both `definition` and `sourceLocation` can be null here. Built-in object types have null definitions, while built-in scalars have definitions that would appear to be default/uninitialized objects